1. Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy
Полученные значения должны быть равны.
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.

In [1]:
import numpy as np
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])
cov = np.mean(zp * ks) - np.mean(zp) * np.mean(ks)
cov

9157.839999999997

In [4]:
np.cov(zp, ks, ddof = 0)

array([[ 3494.64,  9157.84],
       [ 9157.84, 30468.89]])

In [10]:
r1 = np.cov(zp, ks, ddof = 0) / (np.std(zp) * np.std(ks))
r1

array([[0.33866702, 0.88749009],
       [0.88749009, 2.95275283]])

In [9]:
r2 = np.corrcoef(zp, ks)
r2

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

In [14]:
import pandas as pd
a = {
    "zp": [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
    "ks": [401, 574, 874, 919, 459, 739, 653, 902, 746, 832]
}

b = pd.DataFrame(a)

b

zp   ks
0   35  401
1   45  574
2  190  874
3  200  919
4   40  459
5   70  739
6   54  653
7  150  902
8  120  746
9  110  832

In [15]:
b.corr()

zp       ks
zp  1.00000  0.88749
ks  0.88749  1.00000

2. Измерены значения IQ выборки студентов,
обучающихся в местных технических вузах:
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.
Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [16]:
a = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])
M = np.mean(a)
M

118.1

In [17]:
# найдем критерий Стьюдента при а/2 = 2,5% и степени свободы n-1 = 10-1
t = 2.262

In [18]:
# несмещенная дисперсия
a.var(ddof=1)

111.21111111111111

In [21]:
M + t * np.sqrt(a.var(ddof=1)) / np.sqrt(10)

125.64339223691834

In [22]:
M - t * np.sqrt(a.var(ddof=1)) / np.sqrt(10)

110.55660776308164

## [110.556, 125.643]

3.Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,
среднее выборочное составляет 174.2. Найдите доверительный интервал для математического
ожидания с надежностью 0.95.

In [23]:
# найдем критерий Z для a/2 = 2,5%
Z = 1.96
M = 174.2
n = 27
sigma = 5
M + Z * sigma / np.sqrt(n)

176.08601087935276

In [24]:
M - Z * sigma / np.sqrt(n)

172.31398912064722

## [172.31, 176.09]